garden_ai/mlmodel.py

```diff
@@ -141,6 +145,74 @@ def upload_to_model_registry(local_model: LocalModel) -> RegisteredModel:
     return _assemble_metadata(local_model)
 
 
+def stage_model_for_upload(local_model: LocalModel) -> str:
+    """
+    Parameters
+    ----------
+    local_model: the model to be staged into an MLModel directory
+
+    Returns full path of model directory
+    -------
+    """
+    flavor, local_path = local_model.flavor, local_model.local_path
+    try:
+        if flavor == ModelFlavor.SKLEARN.value and pathlib.Path(local_path).is_file:
+            with open(local_path, "rb") as f:
+                loaded_model = pickle.load(f)
+                log_model_variant = mlflow.sklearn.log_model
+        elif flavor == ModelFlavor.TENSORFLOW.value:
+            os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
+            # ignore cpu guard info on tf import require before tf import
+            from tensorflow import keras  # type: ignore
+
+            loaded_model = keras.models.load_model(local_path)
+            log_model_variant = (
+                mlflow.tensorflow.log_model
+            )  # TODO explore artifact path and sigs
+        elif flavor == ModelFlavor.PYTORCH.value and pathlib.Path(local_path).is_file:
+            import torch  # type: ignore
+
+            loaded_model = torch.load(local_path)
+            log_model_variant = mlflow.pytorch.log_model  # TODO explore signatures
+        else:
+            raise ModelUploadException(f"Unsupported model flavor {flavor}")
```

Comment:
This whole first part of the function is copypasta from _push_model_to_registry. But in my next PR where I do the download side, I'll delete `_push_model_to_registry` so we'll be DRY again.